robottelo/test.py

```diff
@@ -389,13 +394,27 @@ def _get_vm_list(cls):
         cls.vm_list = vm_list_string.split(',')
 
     @classmethod
-    def _set_testcase_parameters(cls, savepoint_name,
-                                 raw_file_name, stat_file_name):
+    def _set_testcase_parameters(cls, savepoint_name, raw_file_name,
+                                 stat_file_name, **kwargs):
         # note: set savepoint empty to continue test without restore
         cls.savepoint = conf.properties.get(savepoint_name, '')
         cls.raw_file_name = raw_file_name
         cls.stat_file_name = stat_file_name
 
+        if not kwargs:
+            # for subscription by ak or deletion case, would output
+            # 1. raw data file for ak/del:  raw_file_name
+            # 2. stat data file for ak/del: stat_file_name
+            return
+        else:
+            # for subscription by attach case, would output:
+            # 1. raw data file for register:  reg_raw_file_name
+            # 2. raw data file for attach:    raw_file_name
+            # 3. stat data file for register: reg_stat_file_name
+            # 4. stat data file for attach:   stat_file_name
+            cls.reg_raw_file_name = kwargs.get('raw_reg')
+            cls.reg_stat_file_name = kwargs.get('stat_reg')
```

Comment:
There's a couple ways to re-write this block of code, depending on what's legal. Given this signature:

```python
    @classmethod
    def _set_testcase_parameters(
            cls,
            savepoint_name,
            raw_file_name,
            stat_file_name,
            raw_reg=None,
            stat_reg=None):
```

This could work:

```python
if raw_reg is not None and stat_reg is not None:
    cls.raw_reg = raw_reg
    cls.stat_reg = stat_reg
```

Or if just one value may be provided:

```python
if raw_reg is not None:
    cls.raw_reg = raw_reg
if stat_reg is not None:
    cls.stat_reg = stat_reg
```

Or if `None` are legal values:

```python
cls.raw_reg = raw_reg
cls.stat_reg = stat_reg
```